presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxSplitManager.java

```diff
@@ -57,7 +58,9 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
         TupleDomain<ColumnHandle> predicate = jmxLayout.getConstraint();
 
         //TODO is there a better way to get the node column?
-        JmxColumnHandle nodeColumnHandle = tableHandle.getColumns().get(0);
+        JmxColumnHandle nodeColumnHandle = tableHandle.getColumns().stream()
+                .filter(jmxColumnHandle -> jmxColumnHandle.getColumnName().equals(NODE_COLUMN_NAME))
+                .findFirst().get();
```

Comment:
can there be two of those?
It would be nice to replace `findFirst` with sth like `getOnlyElement` - but java does not have sth like this.
You may want to write collector for that. Not felling strongly about it.